static/js-src/upcoming-page.js

```diff
@@ -0,0 +1,94 @@
+// Start fetching right away.
+const FEATURES_API_URL = '/api/v0/features';
+const CHANNELS_API_URL = '/api/v0/channels';
+const channelsArray = ['stable', 'beta', 'dev'];
+
+const channelsPromise = fetch(CHANNELS_API_URL)
+  .then((res) => res.text())
+  .then((res) => JSON.parse(res.substring(5)));
+
+document.querySelector('.show-blink-checkbox').addEventListener('change', e => {
+  e.stopPropagation();
+  document.querySelector('chromedash-schedule').showBlink = e.target.checked;
+});
+
+const header = document.querySelector('app-header-layout app-header');
+if (header) {
+  header.fixed = false;
+}
+
+async function init() {
+  // Prepare data for chromedash-schedule
+  const channels = await channelsPromise;
+  let featuresPromise = {};
+
+  channelsArray.forEach((channel) => {
+    let queryStringUrl = `${FEATURES_API_URL}?milestone=${channels[channel].version}`;
+    featuresPromise[channel] = fetch(queryStringUrl)
+      .then((res) => res.text())
+      .then((res) => JSON.parse(res.substring(5))); // Ignore XSSI prefix
+  });
+
+  const features = {};
+
+  for (let channel of channelsArray) {
+    features[channel] = await featuresPromise[channel];
+  }
+
+  // Remove the loading sign once the data has been fetched from the APIs
+  document.body.classList.remove('loading');
+
+  channelsArray.forEach((channel) => {
+    channels[channel].components = mapFeaturesToComponents(features[channel].filter(f =>
+      f.browsers.chrome.status.milestone_str === channels[channel].version));
+  });
+
+  const scheduleEl = document.querySelector('chromedash-schedule');
+  scheduleEl.channels = channels;
+
+  window.csClient.getStars().then((starredFeatureIds) => {
+    scheduleEl.starredFeatures = new Set(starredFeatureIds);
+  });
+}
+
+function mapFeaturesToComponents(features) {
+  let set = new Set();
```

Comment:
I don't see where set is used.

Reply:
It seems to be unreferenced even in the chromedash-schedule.js

I have removed it in [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1426/commits/8dd8fa3e89b48b1ab7b2d9ad8981a3956cdabe87) commit.

Reply:
Thanks